dbms/src/Flash/Mpp/ReceivedMessageQueue.cpp

```diff
@@ -59,32 +60,30 @@ std::pair<MPMCQueueResult, ReceivedMessagePtr> ReceivedMessageQueue::pop(size_t
         {
             res = msg_channels_for_fine_grained_shuffle[stream_id]->tryPop(recv_msg);
         }
-        if (recv_msg != nullptr)
+        if (res == MPMCQueueResult::OK)
         {
+            assert(recv_msg != nullptr);
             if (recv_msg->getRemainingConsumers()->fetch_sub(1) == 1)
             {
-                ReceivedMessagePtr original_msg;
-                auto pop_result [[maybe_unused]] = grpc_recv_queue->tryPop(original_msg);
+                auto pop_result [[maybe_unused]] = grpc_recv_queue->dequeue();
```

Comment:
I would like to keep the remaining consumers check, at least in debug mode.

Reply:
ok, added in debug mode.